### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by NotCoffee418** [NotCoffee418/TrinityCreator#91]
For educational purposes.

**Comment by illfated** [comment]
Good idea. I remember noticing "unit tests" was very often part of the added content in the Git Extensions repository. I have not found the time to learn enough about it due to my lack of coding abilities, though.

**Comment by NotCoffee418** [comment]
@illfated It's simple enough. tl;dr it just tries to run methods with a predefined input to see if it gets a desired result.
If for example Sum(1,2) returns 3, the tests succeeds. If it returns anything else it fails and there's a problem with the code.

TCreator is a bit nasty to write these tests for since it's more of a translator than actual data processing with the output (sql) being tricky to validate with automated tests. 
So I'll have to look at best practice and see what i should/shouldn't be doing.
But first I'm trying it in a fresh project to see how that goes. And see if it helps or slows me down.

**Comment by illfated** [comment]
👍 Very good, thanks for the reply. I understand the concept, I just find it a tiny bit troublesome to visualize the actual methods needed for a test in the actual programming languages, like C# or C++ (seeing that my primary experience with coding is the repository TrinityCore/TrinityCore/). Fair enough, I have contributed a little bit in game scripts (with some help) and some database updates (MySQL), but I don't find it tempting to learn more C++ (or C# for that matter), even though I have been advised to try learning more via some python courses to see if I can improve my skills that way. Anyway, I guess it is more a question about what I like to do in my spare time, after coming home from work and just want to do things I already know well enough to be able to contribute. That is mostly why I found it nice to join the Microsoft Docs repo (IT Pro & Windows Server) to offer my skills in English grammar and MarkDown codestyle. Thank you for the useful example, though, it is nice to know more about what goes on in environments I like to follow. 😃
